apis/metal3.io/v1alpha1/bmceventsubscription_validation.go

```diff
@@ -0,0 +1,66 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// bmclog is for logging in this package.
+var bmclog = logf.Log.WithName("bmceventsubscription-validation")
+
+// validateSubscription validates BMCEventSubscription resource for creation
+func (s *BMCEventSubscription) validateSubscription() []error {
+	bmclog.Info("validate create", "name", s.Name)
+	var errs []error
+
+	if s.Spec.HostName == "" {
+		errs = append(errs, fmt.Errorf("HostName cannot be empty"))
+	}
+
+	if s.Spec.Destination == "" {
+		errs = append(errs, fmt.Errorf("Destination cannot be empty"))
+	} else {
+		destinationUrl, err := url.ParseRequestURI(s.Spec.Destination)
+
+		if err != nil {
+			errs = append(errs, fmt.Errorf("Destination is an invalid URL"))
+		} else {
+			if !strings.HasSuffix(destinationUrl.Path, "/") {
+				errs = append(errs, fmt.Errorf("Destination must have a trailing slash"))
```

Comment:
This is a weird requirement, does it come from Redfish?

Reply:
Yes

Reply:
I think you only need to have the `/` in the end in some cases, otherwise this wouldn't work `https://<IP>/RedfishEvents/EventReceiver.php` right?  Only if you are trying the destination as `https://<IP>` you would need the extra `/`

https://developer.hpe.com/blog/the-redfish-event-service/